sunpy/image/tests/test_resample.py

```diff
@@ -39,7 +40,9 @@ def resample_method(aia171_test_map, method):
 
 
 def test_resample_neighbor(aia171_test_map):
-    resample_method(aia171_test_map, 'neighbor')
+    with pytest.warns(SunpyDeprecationWarning,
+                      match='Using "neighbor" as a method for resampling is deprecated.'):
+        resample_method(aia171_test_map, 'neighbor')
```

Comment:
Could you maybe add a test here that ensures that `"nearest"` and `"neighbor"` do give equivalent results? It can then be removed once the deprecation period has passed.